### PR TITLE
Detect unpublished packages

### LINKED ISSUE #1: versionPkg() throws "Cannot read property 'latest' of undefined"
Hi,
While running [npm-name-cli](https://github.com/sindresorhus/npm-name-cli), I got the following error:

```sh
npm-name --version
> 2.3.0

npm-name sunglasses
> TypeError: Cannot read property 'latest' of undefined
>    at versionPkg (/usr/local/lib/node_modules/npm-name-cli/node_modules/squatter/lib/get->package.js:9:36)
>    at squatter (/usr/local/lib/node_modules/npm-name-cli/node_modules/squatter/index.js:42:17)
>    at process._tickCallback (internal/process/next_tick.js:68:7)
```

which points to [the following line of code](https://github.com/sholladay/squatter/blob/master/lib/get-package.js#L9) : 

```js
    const fixed = meta['dist-tags'][version] // version = 'latest'
```
I logged `meta` I and I got the following result: 

```js
{ _id: 'sunglasses',
  _rev: '2-120e55f8b26a7ff42558a6baf3f1d868',
  name: 'sunglasses',
  time:
   { modified: '2016-04-28T00:32:39.237Z',
     created: '2016-04-28T00:32:39.237Z',
     '0.0.0-reserved': '2016-04-28T00:32:39.237Z',
     unpublished:
      { name: 'npm',
        time: '2016-06-10T22:41:14.482Z',
        tags: [Object],
        maintainers: [Array],
        versions: [Array] } },
  _attachments: {} }
```

### PR DESCRIPTION
**Pull request by bokub**
Fixes #1 

I couldn't unpublish 'sunglasses' 😎 , but I found 'leopard' 🐆 to be unpublished too so I added it to the tests

### CODE REVIEW

**Comment by sholladay** [comment]
I wonder what `meta` will look like for a package where a specific version was unpublished, rather than every version. The latter is npm's default, but you can specify a version in `npm unpublish`.

As for how to find a sample, perhaps `left-pad` still has older versions that are unpublished and were never restored.

**Comment by bokub** [comment]
I've just tested, and here is the result: 
(I've unpublished the `0.0.1`)

```js
{
  // ...
  'dist-tags': { latest: '0.0.2' },
  versions:
   { '0.0.2':
      { //...
       }
   },
  time:
   { created: '2018-08-28T17:03:13.524Z',
     '0.0.1': '2018-08-28T17:03:13.692Z',
     modified: '2018-08-28T17:07:04.066Z',
     '0.0.2': '2018-08-28T17:06:04.272Z' },

```

**Comment by sholladay** [comment]
Awesome, that's about what I would have expected.

The reason that matters is that squatter takes a second (currently undocumented) `version` argument. I'm not sure if there is a strong use case for it, but I implemented it so that i could use it in the tests if it turned out that the integration tests were too flakey, since they use the `latest` of real packages and someone could publish a new version that breaks those tests at any time. So far that has never been a problem, but I'm keeping an eye on it.

Anyway, in order to handle the `version` argument correctly, I guess `unpublished()` would have to first resolve the version using `dist-tags` (but handle it not existing) and then check if that version exists in `Object.keys(meta.versions)`. Based on the output you posted in the issue, it seems like when all versions are unpublished, they move `meta.versions` to `meta.time.unpublushed.versions`. So I suppose it would have to handle that, too.

Or to simplify everything, we could get rid of the `version` argument.

**Comment by sholladay** [comment]
As the owner, you can choose to unpublish all versions or only some versions. The default behavior when you run `npm unpublish` is to unpublish all versions, but people can choose to unpublish specific versions if they feel like it (e.g. `0.0.2` had a major security flaw, but `0.0.3` has the fix, and you want to unpublish _only_ `0.0.2` to ensure no one is affected by the security flaw).

So unpublishing depends upon versioning in the same way that publishing does. The only part I'm not super familiar with is how they represent this in the registry data. I'll help investigate that and see if there is a way for us to tell that a specific version was unpublished or whether the entire namespace was unpublished. Give me a few days and I'll look into it!

**Comment by brekk** [comment]
Any follow-up on this? I recently was using `npm-name-cli` which uses `squatter` as a dependency -- I was looking to publish `zark` but as far as I can tell from the registry it is unpublished / squatted upon by the user `~zark`:

```sh
npm-name zark
⠇ Checking name on npmjs.com…
```
Sat and hung for a while, until I hit Ctrl-C and then I saw:
```
^CTypeError: Cannot read property 'latest' of undefined
    at versionPkg (/[...]/.config/yarn/global/node_modules/squatter/lib/get-package.js:9:36)
    at squatter ([...[/.config/yarn/global/node_modules/squatter/index.js:42:17)
    at process._tickCallback (internal/process/next_tick.js:68:
```

I'd love to be able to use `npm-name` / `squatter` to track this status.

**Comment by sholladay** [comment]
I will be fixing this, for sure. Most likely merging this PR as-is. I just need to find the time to verify my assumptions about the registry data are correct. Tentatively thinking I should be able to do that middle of this week. Soon as this is done, I will make a patch release. Thanks for your patience. :)

**Comment by laggingreflex** [comment]
@brekk That error is probably fixed by https://github.com/sindresorhus/npm-name-cli/pull/15

**Comment by ctexxx** [comment]
any updates on this? @sholladay

File: package.json
Comment (sholladay) [inline package.json]: Could you please either undo this change or give me write access to your fork so I can do it for you? (I'll bump the version during the release process.)

**Review by sholladay — CHANGES_REQUESTED** [review]

**Comment by sholladay** [comment]
Hi! I finally got around to this. Ready to merge with that one tweak. Thanks for your patience. :)

**Comment by sindresorhus** [comment]
@bokub Ping

**Comment by bokub** [comment]
Hi Sindre
I can't edit my PR because I've deleted the fork a long time ago 🙃
Maybe @sholladay can make an edit before or after merging? 

If not, I'll make a new PR tomorrow when I'm around a computer

**Comment by sholladay** [comment]
For some reason, GitHub will allow me to merge, but not edit. I presume this is because the fork was deleted.

Don't worry about recreating the fork. What I'll do is squash merge and then amend the commit on master.

Thanks for your work on this!

**Comment by bokub** [comment]
I've duplicated this PR in #5 😊

**Comment by bokub** [comment]
@sholladay Ping 🙋

**Comment by sholladay** [comment]
I merged PR #5. Thanks for your work. I am aiming to do a release next weekend, possibly sooner.

**Comment by sindresorhus** [comment]
@sholladay Reminder about releasing.

**Comment by sholladay** [comment]
This fix has been released. Thanks for your patience, y'all. I've been taking care of my father who is very sick. I'm behind on many things, but OSS keeps me going. :)

**Pull request closed without merging**
